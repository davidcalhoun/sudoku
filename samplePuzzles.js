export const puzzles = [
    {
        name: 'easy1',
        grid: [
            [0, 3, 7, 0, 0, 0, 1, 6, 0],
            [0, 0, 0, 9, 0, 0, 0, 0, 4],
            [9, 0, 0, 1, 3, 6, 0, 0, 0],

            [0, 8, 0, 0, 1, 5, 0, 9, 2],
            [0, 1, 2, 0, 0, 0, 5, 7, 0],
            [5, 7, 0, 2, 4, 0, 0, 1, 0],

            [0, 0, 0, 5, 6, 4, 0, 0, 1],
            [3, 0, 0, 0, 0, 7, 0, 0, 0],
            [0, 6, 4, 0, 0, 0, 9, 8, 0]
        ]
    },
    {
        name: 'hard1',
        grid: [
            [0, 0, 0, 0, 9, 4, 0, 0, 3],
            [0, 6, 8, 0, 1, 0, 0, 0, 9],
            [0, 0, 0, 6, 2, 0, 0, 4, 5],

            [0, 4, 0, 0, 0, 0, 9, 3, 0],
            [0, 0, 0, 0, 0, 0, 0, 0, 0],
            [0, 2, 9, 0, 0, 0, 0, 6, 0],

            [1, 9, 0, 0, 7, 8, 0, 0, 0],
            [2, 0, 0, 0, 6, 0, 7, 9, 0],
            [6, 0, 0, 1, 5, 0, 0, 0, 0]
        ]
    }
];
