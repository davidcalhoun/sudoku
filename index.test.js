import { solve } from './';
import { puzzles } from './samplePuzzles';

describe('sudoku', () => {
    puzzles.forEach(({ name, grid }) => {
        test(name, () => {
            const solution = solve(grid);
            expect(solution).toMatchSnapshot();
        });
    });
});