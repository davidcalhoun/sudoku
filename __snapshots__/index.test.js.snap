// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sudoku easy1 1`] = `
Array [
  Array [
    8,
    3,
    7,
    4,
    5,
    2,
    1,
    6,
    9,
  ],
  Array [
    1,
    2,
    6,
    9,
    7,
    8,
    3,
    5,
    4,
  ],
  Array [
    9,
    4,
    5,
    1,
    3,
    6,
    8,
    2,
    7,
  ],
  Array [
    6,
    8,
    3,
    7,
    1,
    5,
    4,
    9,
    2,
  ],
  Array [
    4,
    1,
    2,
    6,
    8,
    9,
    5,
    7,
    3,
  ],
  Array [
    5,
    7,
    9,
    2,
    4,
    3,
    6,
    1,
    8,
  ],
  Array [
    2,
    9,
    8,
    5,
    6,
    4,
    7,
    3,
    1,
  ],
  Array [
    3,
    5,
    1,
    8,
    9,
    7,
    2,
    4,
    6,
  ],
  Array [
    7,
    6,
    4,
    3,
    2,
    1,
    9,
    8,
    5,
  ],
]
`;

exports[`sudoku hard1 1`] = `
Array [
  Array [
    5,
    7,
    2,
    8,
    9,
    4,
    6,
    1,
    3,
  ],
  Array [
    4,
    6,
    8,
    3,
    1,
    5,
    2,
    7,
    9,
  ],
  Array [
    9,
    1,
    3,
    6,
    2,
    7,
    8,
    4,
    5,
  ],
  Array [
    7,
    4,
    1,
    5,
    8,
    6,
    9,
    3,
    2,
  ],
  Array [
    3,
    5,
    6,
    9,
    4,
    2,
    1,
    8,
    7,
  ],
  Array [
    8,
    2,
    9,
    7,
    3,
    1,
    5,
    6,
    4,
  ],
  Array [
    1,
    9,
    4,
    2,
    7,
    8,
    3,
    5,
    6,
  ],
  Array [
    2,
    8,
    5,
    4,
    6,
    3,
    7,
    9,
    1,
  ],
  Array [
    6,
    3,
    7,
    1,
    5,
    9,
    4,
    2,
    8,
  ],
]
`;
